### DISCUSSION TITLE
com.auth0.jwk.SigningKeyNotFoundException: JWT Token keyID doesn't match the configured keyID.

### DISCUSSION THREAD

**Discussion by liujuanchi** [open-metadata/OpenMetadata#18573] [Q&A] [answered]
when I configure the snowflake connection on UI, got this alert:

test-connection-Snowflake-RGTZzAgQ Failed to trigger workflow due to airflow API returned Internal Server Error and response {"error": "Error running automation workflow due to [An exception with message [com.auth0.jwk.SigningKeyNotFoundException: JWT Token keyID doesn't match the configured keyID. This usually happens if you didn't configure proper publicKeyUrls under authentication configuration.] was thrown while processing request.] "}

any one have insight? much appreciate!

**Comment by Prajwal214** [comment, marked as answer]
The error you're encountering, "JWT Token keyID doesn't match the configured keyID," indicates a mismatch between the keyID in your JWT token and the one configured in OpenMetadata. This typically happens when the public key URLs are not properly configured. Here are steps to resolve this issue:

1. Verify Key ID Consistency: Ensure that the keyID in your JWT token configuration matches the keyID used when generating the token. The keyID should be consistent across the configuration and the token generation process.

2. Update Public Key URLs: Double-check the public key URLs in your OpenMetadata configuration. Ensure that they are correctly specified and accessible. The URLs should point to the location where your authentication server's public keys are hosted ([source](https://docs.open-metadata.org/latest/deployment/security/enable-jwt-tokens)).

3. Regenerate Token: After updating the keyID and public key URLs, revoke the existing token and generate a new one in the OpenMetadata UI. This ensures the new token uses the correct keyID.

4. Check Airflow Configuration: Ensure that the Airflow configuration is updated to use the new JWT token. This involves updating the connection details in the Airflow UI under Admin -> Connections for openmetadata_conn_id.

**Comment by ognjen-it** [comment]
I had the same problem when my IDP (Keycloak) had a self-signed cert.
